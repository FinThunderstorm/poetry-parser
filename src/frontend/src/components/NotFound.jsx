import React from 'react'
import { Paper, Stack, Typography } from '@mui/material'

/** NotFound is used if given page is not found
 */
const NotFound = () => (
    <Paper elevation={6} sx={{ m: 1, mt: 3, p: 2 }}>
        <Stack spacing={2} alignItems="center">
            <Typography variant="button" sx={{ fontSize: '1.5rem' }}>
                404 NOT FOUND
            </Typography>
            <Typography>
                Our pear eating robots was too lazy and could not catch page
                behind given url.
            </Typography>
        </Stack>
    </Paper>
)

export default NotFound
