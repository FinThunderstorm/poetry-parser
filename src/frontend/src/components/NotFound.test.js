/* eslint-disable react/prop-types */
import React from 'react'
import renderer from 'react-test-renderer'
import '@testing-library/jest-dom/extend-expect'
import { render, screen } from '@testing-library/react'

import NotFound from './NotFound'

describe('NotFound', () => {
    test('NotFound renderes correctly', () => {
        render(<NotFound />)

        const title = screen.getByText('404 NOT FOUND', {
            exact: false,
        })
        const info = screen.getByText(
            'Our pear eating robots was too lazy and could not catch page behind given url.',
            {
                exact: false,
            }
        )
        expect(title).toBeDefined()
        expect(info).toBeDefined()

        const tree = renderer.create(<NotFound />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})
