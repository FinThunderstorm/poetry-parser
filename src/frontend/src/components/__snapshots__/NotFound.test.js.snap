// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound NotFound renderes correctly 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation6 css-crdtso-MuiPaper-root"
>
  <div
    className="css-1y48f0p-MuiStack-root"
  >
    <span
      className="MuiTypography-root MuiTypography-button css-1sqoee-MuiTypography-root"
    >
      404 NOT FOUND
    </span>
    <p
      className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      Our pear eating robots was too lazy and could not catch page behind given url.
    </p>
  </div>
</div>
`;
